scheduler/reconcile.go

```diff
@@ -424,10 +424,10 @@ func (a *allocReconciler) computeGroup(group string, all allocSet) bool {
 
 	// The fact that we have destructive updates and have less canaries than is
 	// desired means we need to create canaries
-	numDestructive := len(destructive)
 	strategy := tg.Update
 	canariesPromoted := dstate != nil && dstate.Promoted
-	requireCanary := numDestructive != 0 && strategy != nil && len(canaries) < strategy.Canary && !canariesPromoted
+	requireCanary := (len(destructive) != 0 || (len(untainted) == 0 && len(migrate)+len(lost) != 0)) &&
+		strategy != nil && len(canaries) < strategy.Canary && !canariesPromoted
```

Comment:
This is semi-related band-aid that we'll probably need to investigate further.  The code here determines if canaries are needed by checking if we have any destructive update.  However, if all allocations are dead (because the nodes are lost), `len(destructive)` will be 0.  I changed the condition to account for such scenario.